simapp/app_v2.go

```diff
@@ -246,8 +247,7 @@ func NewSimApp(
 	app.App = appBuilder.Build(logger, db, traceStore, baseAppOptions...)
 
 	if err := app.App.BaseApp.SetStreamingService(appOpts, app.appCodec, app.kvStoreKeys()); err != nil {
-		logger.Error("failed to load state streaming", "err", err)
```

Comment:
I understand why we should do that in the SDK itself, but in an app it's actually clearer to have a error logged and exiting nicely.
The error message due to the panic is less friendly that logging an error.
It was panicking in previous versions but that's why it's like this now.